Code/GraphMol/SmilesParse/SmartsWrite.cpp

```diff
@@ -579,6 +585,11 @@ std::string getNonQueryAtomSmarts(const QueryAtom *qatom) {
       res << "+" << qatom->getFormalCharge();
     }
   }
+  int mapNum;
```

Comment:
yeah, I've never been sure about that one. But if we're going to change that behavior it's likely a different issue/PR